joule-labs/joule-extension
# Add Lightning Loop functionality

Closes #225 

### Description

This PR attempts to address inbound liquidity issues by adding support for "Looping Out/In" to an off-chain address. The default address will be the lnd wallet address. The plan is to add a loop-http library similar to the lnd-http lib implementation. Will update as necessary pending changes to the lightning labs loop repo.

LND 0.7.0
Loop 2.2
* The branch below has the changes needed to build Loop In API changes
  you will also need to run ./gen_protos.sh in the looprpc directory
https://github.com/lightninglabs/loop/pull/74

### Steps to Test

1. Install Loopd 
2. View error handling for incorrect URL / Set correct Loop URL
3. View loop out terms (popover on advance settings buttons)
4. View Loop Out/In Quote
5. Attempt Loop Out/In to off-chain address (default LND wallet)
6. Use 'Loop Monitor' in terminal to check status
7. Default/Advanced User settings
8. Verify password before quote -> loop out/in

### Screenshot
![image](https://user-images.githubusercontent.com/13033037/62595200-6aaaaf00-b8ab-11e9-8ff1-7a2d7ac5835d.png)
